src/Microsoft.Health.Dicom.Core.UnitTests/Features/CustomTag/CustomTagCacheTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Core.UnitTests.Features.CustomTag
+{
+    public class CustomTagCacheTests
+    {
+        private static readonly TimeSpan DefaultExpirationInterval = TimeSpan.FromSeconds(30);
+
+        [Fact]
+        public async Task GivenUninitializedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldInitialize()
+        {
+            long expectedVersion = 1;
+            List<CustomTagEntry> expectedResult = new List<CustomTagEntry>();
+            expectedResult.Add(new CustomTagEntry() { Version = expectedVersion });
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult, expectedResult);
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, DefaultExpirationInterval);
+            CustomTagList customTags = await customTagCache.GetCustomTagsAsync();
+            Assert.Equal(expectedResult, customTags.CustomTags);
+            Assert.NotNull(customTagCache.LastRefreshTime);
+        }
+
+        [Fact]
+        public async Task GivenOutdatedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldRefreshCache()
+        {
+            long expectedVersion1 = 1, expectedVersion2 = 2;
+            List<CustomTagEntry> expectedResult1 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(version: expectedVersion1) };
+            List<CustomTagEntry> expectedResult2 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(version: expectedVersion2) };
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult1, expectedResult2);
+
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, TimeSpan.FromSeconds(1));
+            CustomTagList customTags1 = await customTagCache.GetCustomTagsAsync();
+            DateTimeOffset lastRefreshTime = customTagCache.LastRefreshTime.Value;
+            await Task.Delay(TimeSpan.FromSeconds(2));
+
```

Comment:
Delaying for seconds could make our tests run very long, can we do shorter delays?

Reply:
would 1 second be better?